cmds/hwclock/hwclock.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// hwclock sets or reads the hwclock in UTC format.
+//
+// Synopsis:
+//     hwclock [-w]
+//
+// Description:
+//     It prints the current hwclock time in UTC if called without any flags.
+//     It sets the hwclock to the system clock in UTC if called with -w.
+//
+// Options:
+//     -w: set hwclock to system clock in UTC
+//
+// Author:
+//     claudiozumbo@gmail.com.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+const startYear = 1900
```

Comment:
That's what I thought as well but it runs out that the tm struct does not follow epoch:

https://pubs.opengroup.org/onlinepubs/7908799/xsh/time.h.html

"int    tm_year  years since 1900"

So this is used to sum/subtract when reading/writing RTC.

Reply:
wow. Thanks I did not know that.